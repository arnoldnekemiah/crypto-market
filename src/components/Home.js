import { DateTime } from 'luxon';
import { SiCoinmarketcap } from 'react-icons/si';
import { BsFillMicFill } from 'react-icons/bs';
import { IoSettingsSharp } from 'react-icons/io5';
import Navigation from './Navigation';

const dt = DateTime.now();
const date = dt.toFormat('dd LLL yy');
const Home = () => (
  <div>
    <header className="row">
      <div className="first">
        {date}
      </div>
      <p>Top 100 Tokens</p>
      <div className="row dummyIcons">
        <BsFillMicFill />
        <IoSettingsSharp />
      </div>
    </header>
    <div id="home" className="row">
      <SiCoinmarketcap id="logo" />
      <div className="column">
        <h1>Crypto Market Rankings Cap</h1>
      </div>
    </div>
    <Navigation />
  </div>
);
export default Home;
