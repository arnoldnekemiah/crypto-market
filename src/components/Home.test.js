import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Home from './Home';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  crypto: {
    cryptoArray: [],
    isLoading: false,
    error: null,
  },
});

describe('Home', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
