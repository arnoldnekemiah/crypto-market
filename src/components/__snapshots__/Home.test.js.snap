// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <header
    className="row"
  >
    <div
      className="first"
    >
      06 Sep 23
    </div>
    <p>
      Top 100 Tokens
    </p>
    <div
      className="row dummyIcons"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 176a80 80 0 1080 80 80.24 80.24 0 00-80-80zm172.72 80a165.53 165.53 0 01-1.64 22.34l48.69 38.12a11.59 11.59 0 012.63 14.78l-46.06 79.52a11.64 11.64 0 01-14.14 4.93l-57.25-23a176.56 176.56 0 01-38.82 22.67l-8.56 60.78a11.93 11.93 0 01-11.51 9.86h-92.12a12 12 0 01-11.51-9.53l-8.56-60.78A169.3 169.3 0 01151.05 393L93.8 416a11.64 11.64 0 01-14.14-4.92L33.6 331.57a11.59 11.59 0 012.63-14.78l48.69-38.12A174.58 174.58 0 0183.28 256a165.53 165.53 0 011.64-22.34l-48.69-38.12a11.59 11.59 0 01-2.63-14.78l46.06-79.52a11.64 11.64 0 0114.14-4.93l57.25 23a176.56 176.56 0 0138.82-22.67l8.56-60.78A11.93 11.93 0 01209.94 26h92.12a12 12 0 0111.51 9.53l8.56 60.78A169.3 169.3 0 01361 119l57.2-23a11.64 11.64 0 0114.14 4.92l46.06 79.52a11.59 11.59 0 01-2.63 14.78l-48.69 38.12a174.58 174.58 0 011.64 22.66z"
        />
      </svg>
    </div>
  </header>
  <div
    className="row"
    id="home"
  >
    <svg
      fill="currentColor"
      height="1em"
      id="logo"
      role="img"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M20.738 14.341c-.419.265-.912.298-1.286.087-.476-.27-.738-.898-.738-1.774v-2.618c0-1.264-.5-2.164-1.336-2.407-1.416-.413-2.482 1.32-2.882 1.972l-2.498 4.05v-4.95c-.028-1.14-.398-1.821-1.1-2.027-.466-.135-1.161-.081-1.837.953l-5.597 8.987A9.875 9.875 0 0 1 2.326 12c0-5.414 4.339-9.818 9.672-9.818 5.332 0 9.67 4.404 9.67 9.818.004.018.002.034.003.053.05 1.049-.29 1.883-.933 2.29zm3.08-2.34-.001-.055C23.787 5.353 18.497 0 11.997 0 5.48 0 .177 5.383.177 12c0 6.616 5.303 12 11.82 12 2.991 0 5.846-1.137 8.037-3.2.435-.41.46-1.1.057-1.541a1.064 1.064 0 0 0-1.519-.059 9.56 9.56 0 0 1-6.574 2.618c-2.856 0-5.425-1.263-7.197-3.268l5.048-8.105v3.737c0 1.794.696 2.374 1.28 2.544.584.17 1.476.054 2.413-1.468.998-1.614 2.025-3.297 3.023-4.88v2.276c0 1.678.672 3.02 1.843 3.68 1.056.597 2.384.543 3.465-.14 1.312-.828 2.018-2.354 1.944-4.193z"
      />
    </svg>
    <div
      className="column"
    >
      <h1>
        Crypto Market Rankings Cap
      </h1>
    </div>
  </div>
  <nav
    id="coinNav"
  >
    <form
      id="form"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Filter by Rank or Name"
        type="text"
        value=""
      />
    </form>
    <ul
      id="coinGrid"
    />
  </nav>
</div>
`;
